pkg/controllers/consistency/nodeshape.go

```diff
@@ -18,55 +18,37 @@ import (
 	"context"
 	"fmt"
 
-	"github.com/samber/lo"
 	v1 "k8s.io/api/core/v1"
-	"k8s.io/apimachinery/pkg/types"
-	"sigs.k8s.io/controller-runtime/pkg/client"
 
 	"github.com/aws/karpenter-core/pkg/apis/v1alpha5"
 	"github.com/aws/karpenter-core/pkg/cloudprovider"
 )
 
 // NodeShape detects nodes that have launched with 10% or less of any resource than was expected.
 type NodeShape struct {
-	kubeClient client.Client
-	provider   cloudprovider.CloudProvider
+	provider cloudprovider.CloudProvider
 }
 
-func NewNodeShape(kubeClient client.Client, provider cloudprovider.CloudProvider) Check {
+func NewNodeShape(provider cloudprovider.CloudProvider) Check {
 	return &NodeShape{
-		kubeClient: kubeClient,
-		provider:   provider,
+		provider: provider,
 	}
 }
 
-func (n *NodeShape) Check(ctx context.Context, node *v1.Node) ([]Issue, error) {
-	// ignore nodes that are deleting
-	if !node.DeletionTimestamp.IsZero() {
+func (n *NodeShape) Check(ctx context.Context, node *v1.Node, machine *v1alpha5.Machine) ([]Issue, error) {
+	// ignore machines that are deleting
+	if !machine.DeletionTimestamp.IsZero() {
 		return nil, nil
 	}
-	// and nodes that haven't initialized yet
-	if node.Labels[v1alpha5.LabelNodeInitialized] != "true" {
+	// and machines that haven't initialized yet
+	if !machine.StatusConditions().GetCondition(v1alpha5.MachineInitialized).IsTrue() {
 		return nil, nil
 	}
-	provisioner := &v1alpha5.Provisioner{}
-	if err := n.kubeClient.Get(ctx, types.NamespacedName{Name: node.Labels[v1alpha5.ProvisionerNameLabelKey]}, provisioner); err != nil {
-		// provisioner is missing, node should be removed soon
-		return nil, client.IgnoreNotFound(err)
-	}
-	instanceTypes, err := n.provider.GetInstanceTypes(ctx, provisioner)
-	if err != nil {
-		return nil, err
-	}
-	instanceType, ok := lo.Find(instanceTypes, func(it *cloudprovider.InstanceType) bool { return it.Name == node.Labels[v1.LabelInstanceTypeStable] })
-	if !ok {
-		return []Issue{Issue(fmt.Sprintf("instance type %q not found", node.Labels[v1.LabelInstanceTypeStable]))}, nil
-	}
 	var issues []Issue
-	for resourceName, expectedQuantity := range instanceType.Capacity {
+	for resourceName, expectedQuantity := range machine.Status.Capacity {
```

Comment:
Possible to reason about just the machine here, and not the node?

Reply:
I'm not sure that we can here because we are relying on the reported capacity vs. the expected capacity, which we don't sync between the machine and the node